packages/fx-core/src/plugins/resource/apim/managers/apimManager.ts

```diff
@@ -181,4 +223,84 @@ export class ApimManager {
 
     await apimService.addApiToProduct(resourceGroupName, apimServiceName, productId, apiId);
   }
+
+  public async generateArmTemplates(
+    solutionConfig: AzureSolutionSettings
+  ): Promise<ScaffoldArmTemplateResult> {
+    const bicepTemplateDir = path.join(getTemplatesFolder(), ApimPathInfo.BicepTemplateRelativeDir);
+
+    const handleBarsContext = {
+      Plugins: solutionConfig.activeResourcePlugins,
+    };
+
+    const provisionModuleContentResult = await generateBicepFiles(
+      path.join(bicepTemplateDir, ApimPathInfo.ProvisionModuleTemplateFileName),
+      handleBarsContext
+    );
+    if (provisionModuleContentResult.isErr()) {
+      throw provisionModuleContentResult.error;
+    }
+
+    const configurationModuleContentResult = await generateBicepFiles(
+      path.join(bicepTemplateDir, ApimPathInfo.ConfigurationModuleTemplateFileName),
+      handleBarsContext
+    );
+    if (configurationModuleContentResult.isErr()) {
+      throw configurationModuleContentResult.error;
+    }
+
+    const inputParameterContentResult = await generateBicepFiles(
+      path.join(bicepTemplateDir, Bicep.ParameterOrchestrationFileName),
+      handleBarsContext
+    );
+    if (inputParameterContentResult.isErr()) {
+      throw inputParameterContentResult.error;
+    }
+
+    const moduleOrchestrationContentResult = await generateBicepFiles(
```

Comment:
If there's no need to compile, we can skip the compile step.

Reply:
Fixed.